lib/util.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+
+var BBPromise = require('bluebird');
+var util = require('util');
+
+
+/**
+ * Error instance wrapping HTTP error responses
+ */
```

Comment:
This is about the third implementation of this (others: preq, restbase). Should we export it from preq or restbase, or create a httperror package?